api/utils/token_validation.py

```diff
@@ -0,0 +1,19 @@
+from rest_framework.authentication import get_authorization_header
+from users.models import TokenBlacklist
+from rest_framework import permissions, status
+from rest_framework.response import Response
+
+
+class TokenAllowedPermission(permissions.BasePermission):
+
+    def has_permission(self, request, view):
+        auth = get_authorization_header(request).split()
+        token = auth[1]
+        try:
+            blacklisted = TokenBlacklist.objects.get(token=token)
+            if blacklisted != []:
+                return False
```

Comment:
You could also try to reduce the number of return statements per method or function

Reply:
done